internal/hcsoci/create.go

```diff
@@ -278,6 +280,24 @@ func CreateContainer(ctx context.Context, createOptions *CreateOptions) (_ cow.C
 		if err != nil {
 			return nil, r, err
 		}
+
+		if coi.HostingSystem.OS() == "windows" {
+			log.G(ctx).Debug("redirecting container HvSocket for WCOW")
+			props, err := c.PropertiesV2(ctx, hcsschema.PTSystemGUID)
+			if err != nil {
+				return nil, r, fmt.Errorf("query created container properties failed: %w", err)
+			}
+			containerSystemGUID, err := guid.FromString(props.SystemGUID)
```

Comment:
Would this value of `containerSystemGUID` be same as the return value of running `HCSIDToGUID(coi.actualID)`?

Reply:
yeah, they are the same.

Reply:
The function, really, is just a convenience/alternative way to get the VMID. The recommended way would be to query HCS.

Reply:
Hmm, then can the function just call `HcsGetComputeSystemProperties` and get this ID? My only concern is that something can change in HCS and then our implementation of generating this ID won't be correct anymore, so better to use the same implementation at both places.

Reply:
as discussed offline, moving this to internal/tools. I'm a bit hesitant making opening additional handles to compute system to query for properties.